python/pyspark/sql/connect/group.py

```diff
@@ -262,8 +264,44 @@ def applyInPandas(
 
     applyInPandas.__doc__ = PySparkGroupedData.applyInPandas.__doc__
 
-    def applyInPandasWithState(self, *args: Any, **kwargs: Any) -> None:
-        raise NotImplementedError("applyInPandasWithState() is not implemented.")
+    def applyInPandasWithState(
+            self,
+            func: "PandasGroupedMapFunctionWithState",
+            outputStructType: Union[StructType, str],
+            stateStructType: Union[StructType, str],
+            outputMode: str,
+            timeoutConf: str
+    ) -> "DataFrame":
+        from pyspark.sql.connect.udf import UserDefinedFunction
+        from pyspark.sql.connect.dataframe import DataFrame
+
+        udf_obj = UserDefinedFunction(
+            func,
+            returnType=outputStructType,
+            evalType=PythonEvalType.SQL_GROUPED_MAP_PANDAS_UDF_WITH_STATE,
+        )
+
+        output_schema: str = outputStructType.json() if isinstance(outputStructType, StructType)\
```

Comment:
I'm not very sure if we could do like this? In `UserDefinedFunction` above the `returnRype` is of the same type, and it handles it differently 
https://github.com/apache/spark/blob/007a42b8511b005aabdf45955e504d1f71eb1d25/python/pyspark/sql/connect/udf.py#L101
and then
https://github.com/apache/spark/blob/007a42b8511b005aabdf45955e504d1f71eb1d25/python/pyspark/sql/connect/expressions.py#L595-L598
Should we also follow this pattern? 

Maybe @ueshin has more context?

Reply:
Good idea to port tests tests as well. We need to add connect version of `test_pandas_grouped_map_with_state.py`

Reply:
@WweiL The return type of udf is a bit different. Here I am following the way how [DataStreamReader handles schema](https://github.com/apache/spark/blob/007a42b8511b005aabdf45955e504d1f71eb1d25/python/pyspark/sql/connect/streaming/readwriter.py#L61). The server will parse the schema.

Reply:
@rangadi Added versionchanged